### PR TITLE
lint: Remove temp variable in the 'for' loop

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Since golang 1.22+, temp variable in the for loop can be removed. There is new linter copyloopvar in latest golangci-lint, however, there are a lot of false positive now, so probably after a few versions, we can enable it in .golangci.yaml.

### Reference

(copied the below comment from Tobias)

> For future reference: more details on the for loop behavior change in https://tip.golang.org/doc/go1.22#language and https://tip.golang.org/wiki/RangefuncExperiment

### CODE REVIEW

**Review by tklauser — APPROVED** [review]

**Comment by tklauser** [comment]
For future reference: more details on the for loop behavior change in https://tip.golang.org/doc/go1.22#language and https://tip.golang.org/wiki/RangefuncExperiment

File: pkg/datapath/linux/routing/migrate.go [hunk 1]
```diff
@@ -509,14 +509,13 @@ func (m *migrator) copyRoutes(routes []netlink.Route, from, to int) (revert.Reve
 			continue
 		}
 
-		route := r
-		route.Table = to
```
Comment (gentoo-root) [inline pkg/datapath/linux/routing/migrate.go:513]: The loop above has the same pattern.

**Review by pippolo84 — APPROVED** [review]
Thanks! :100:

**Review by joamaki — APPROVED** [review]

**Review by gentoo-root — APPROVED** [review]

**Review by youngnick — APPROVED** [review]

**Pull request merged**